src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -1807,6 +1877,160 @@ public override TypeKind TypeKind
             }
         }
 
+        private bool IsExtensionType([NotNullWhen(true)] out MethodDefinitionHandle? foundMarkerMethod)
+        {
+            var moduleSymbol = this.ContainingPEModule;
+            var module = moduleSymbol.Module;
+            foundMarkerMethod = null;
+
+            var isByRefLike = module.HasIsByRefLikeAttribute(_handle);
+            if (!isByRefLike)
+            {
+                return false;
+            }
+
+            foreach (var field in module.GetFieldsOfTypeOrThrow(_handle))
+            {
+                if ((module.GetFieldDefFlagsOrThrow(field) & FieldAttributes.Static) == 0)
+                {
+                    return false;
+                }
+            }
+
+            try
+            {
+                foreach (var methodHandle in module.GetMethodsOfTypeOrThrow(_handle))
+                {
+                    var methodName = module.GetMethodDefNameOrThrow(methodHandle);
+                    if (methodName == WellKnownMemberNames.ExtensionMarkerMethodName)
+                    {
+                        if (foundMarkerMethod != null)
+                        {
+                            return false;
+                        }
+
+                        foundMarkerMethod = methodHandle;
+                    }
+                }
+
+                return foundMarkerMethod is not null;
```

Comment:
>foundMarkerMethod [](http://example.com/codeflow?start=23&length=17)

Are we going to hide this method from consumers?

Reply:
Yes, given that we'll be hiding the marker method in source symbols (I agree with your recommendation) then I think it makes sense to also hide in PE symbols.